### DISCUSSION TITLE
how to use github token?

### DISCUSSION THREAD

**Discussion by netskink** [zopencommunity/meta#273] [General]
Hello

How to use a github token to avoid API throttle?

I generated a classic token with repo access and did the following based upon `zopen upgrade` error message.

```
$ export ZOPEN_GIT_OUATH_TOKEN="xxxxx"
```

However after attempting the upgrade process, it failed again with the same message.  I don't see any reference in the wiki docs or in issues for meta to specify how to setup the token.  I'm also curious if it needs to be an OAUTH token even though the help message link points to the fine or classic token page.

# Errata

I tried to use the `screenport` port via

```
$ zopen install screenport
$ export PATH=$PATH:~/zopen/dev/screen/bin
$ screen
coredump
```

Figured perhaps the coredump/segmentation fault was because my existing zopen installation is out of date.  Attempted to do an upgrade and that failed.

```
$ cd ~/zopen/dev
$ zopen upgrade
```

Initially worked but eventually failed.  It gave this message:

```
$ zopen upgrade
***WARNING: Run zopen init to configure the install location. Using current working directory...
***WARNING: Setting ZOPEN_GIT_OAUTH_TOKEN is recommended to ensure that you do not hit the GitHub API cap. See --help for more details.
***ERROR: API rate limit exceeded for github.
1. Generate a token (https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/creating-a-personal-access-token).
2. Set ZOPEN_GIT_OAUTH_TOKEN and rerun
```

I used the link provided to setup an access token, exported the env variable, and tried again. It did not seem to recognize the env variable.

```
$ export ZOPEN_GIT_OUATH_TOKEN="xxxxx"
$ zopen upgrade
```

Same error message as before.

**Comment by netskink** [comment]
Ahh, my bad. typo on env variable.

ZOPEN_GIT_OAUTH_TOKEN="xxxxx"

It all works now.

**Comment by v1gnesh** [comment]
Seeing the search results from [this page](https://github.com/ZOSOpenTools/meta/commit/d0fb47842676ecbb35204d74505890467a8012a1), I think you need `ZOPEN_GITHUB_OAUTH_TOKEN` instead of `ZOPEN_GIT_OAUTH_TOKEN`

**Reply by v1gnesh** [reply]
... at least at some point in the future 
[ changed the token reading var name #221 ](https://github.com/ZOSOpenTools/meta/pull/221)

**Reply by netskink** [reply]
yes, I just noticed that.  LOL, i was looking at boot/meta/lib/zopen-install to see the environment variable check and then noticed I had a typo.  Sigh.  Thanks though.  I appreciate it.
